### PR TITLE
Appraisal Letter Pdf Generation

### PR DESCRIPTION
**Pull request by srivastava-jyoti**
Targets #ISSUE_ID

## Breakdown & Estimates 

**Expected Time Range:**

**Actual Time:** 

## Feedback Cycle

**Cycle Count: 0**

## Checklist:

- [ ] The PR title follows this syntax: <#IssueID> \<PR Title>
- [ ] I have linked the issue id in the PR description.
- [ ] I have performed a self-review of my own code.
- [ ] I have added comments on my code changes where required.

### CODE REVIEW

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, let's make these changes whenever you get a chance

File: Modules/Salary/Emails/SendAppraisalLetterMail.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $this->employee = $employee;
+        $this->pdf = $pdf;
+    }
+
+    public function build()
+    {
+        $ccEmails = array_map('trim', explode(',', $this->employee['ccemails']));
+
+        return $this->from('hr@coloredcow.com', 'Mohit Sharma')
```
[the remark this answers is not in the document]
Comment (srivastava-jyoti) [reply Modules/Salary/Emails/SendAppraisalLetterMail.php:29]: Done

File: Modules/Salary/Http/Controllers/SalaryController.php [hunk 2]
```diff
@@ -64,6 +68,33 @@ public function storeOrUpdateSalary(Request $request, Employee $employee)
         $currentSalaryObject->commencement_date = $request->commencementDate;
         $currentSalaryObject->save();
 
+        $salaryService = new SalaryCalculationService($request->grossSalary);
+        $data = $salaryService->sendAppraisalLetterMail($request, $employee);
+
+        $appraisalData = $salaryService->appraisalLetterData($request, $employee);
+        $pdf = $this->showAppraisalLetterPdf($appraisalData);
+        Mail::to($data['employeeEmail'])->send(new SendAppraisalLetterMail($data, $pdf->inline($data['employeeName'] . '.pdf')));
```
[the remark this answers is not in the document]
Comment (srivastava-jyoti) [reply Modules/Salary/Http/Controllers/SalaryController.php]: done

File: Modules/User/Resources/views/profile/show-basic-details.blade.php [hunk 3]
```diff
@@ -81,6 +85,10 @@
                 <label class="font-weight-bold" for="">Email:</label>
                 <span>{{ $user->email }}</span>
             </div>
+            <div class="form-group">
+                <label class="font-weight-bold" for="">Address:</label>
+                {{-- <span>{{ optional($user->profile->address) }}</span> --}}
+            </div>
```
Comment (GauravGusain98) [inline Modules/User/Resources/views/profile/show-basic-details.blade.php:91]: Why line number 90 is commented?

File: Modules/Salary/Config/config.php [hunk 4]
```diff
@@ -32,4 +32,9 @@
             'employer_esi_limit' => 'Employer ESI Limit',
         ],
     ],
+
+    'default' => [
+        'email' => env('HR_DEFAULT_FROM_EMAIL', 'hr@coloredcow.com'),
+        'name' => env('HR_DEFAULT_FROM_NAME', 'Mohit Sharma'),
+    ],
```
Comment (GauravGusain98) [inline Modules/Salary/Config/config.php:39]: update this

**Review by GauravGusain98 — APPROVED** [review]

**Review by P4NK4J — APPROVED** [review]

**Pull request merged**